nshlib/nsh_command.c

```diff
@@ -244,9 +244,10 @@ static const struct cmdmap_s g_cmdmap[] =
 
 #ifdef CONFIG_NET
 # ifndef CONFIG_NSH_DISABLE_IFCONFIG
-  { "ifconfig", cmd_ifconfig, 1, 11,
-    "[interface [<ip-address>|dhcp]] [dr|gw|gateway <dr-address>] "
-    "[netmask <net-mask>] [dns <dns-address>] [hw <hw-mac>]" },
+  { "ifconfig", cmd_ifconfig, 1, 12,
```

Comment:
BTW, why 11 is changed to 12?

Reply:
Because previously `[address_family]` is not taken into count (which is needed for IPv6 and optional for IPv4), and we may have the longest command like `ifconfig eth0 inet6 fc00::2 gw fc00::1 prefixlen 64 dns 2001:4860:4860::8888 hw aa:bb:cc:dd:ee:ff`